Merge UI Work with Master

GraemeRMorgan opened the pull request:
This has all of the navigation bar and updated CSS, as well as Bootstrap integration in the skema.html

GraemeRMorgan committed:
Bootstrap

I've added Bootstrap to Eric's original code.

GraemeRMorgan committed:
Side Navigation Working

Side menu navigation has been implemented with some dummy buttons and search bars.

ErikSearle commented on app/templates/info.html:
```diff
@@ -0,0 +1,24 @@
+    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy" crossorigin="anonymous"></script>
+
```
If we download and include these js and css files in our project, we can use flask-assets to combine them all into a single minifyed css file and a single minifyed js file, which we can import in a single line in our html to make this look much cleaner. This will also reduce our dependencies on these external urls which protects us from a scenario where these sites are down or the version is updated and it makes our site blow up.

ErikSearle commented on app/templates/info.html:
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
```
We shouldn't need a dir tag anymore. It has been deprecated as css now covers this. https://www.computerhope.com/jargon/h/html-dir-tag.htm

ErikSearle commented on app/templates/info.html:
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
```
What is the end game purpose of this info.html file?

GraemeRMorgan replied:
This was something else I was working on. That can go away.

ErikSearle commented on app/templates/skema.html:
```diff
@@ -1,2 +1,141 @@
+          </div>
+        </li>
+        <li>
+          <div class="form-check">
```
This is a good prototype, but once we start serving this html from the flask app, we'll be able to cut down on a lot of this code by using jinja tags. For example, the repetition of the next 4 sections can be dealt with in a jinja for-loop.

GraemeRMorgan replied:
Sweet. I'm looking into jinja templates.

ErikSearle commented on app/templates/skema.html:
```diff
@@ -1,2 +1,141 @@
+  <!-- Bootstrap Js CDN -->
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+
+  <script type="text/javascript">
```
Ideally this should be in a js file, and not written directly into the html. This isn't a big issue obviously as it's only 4 lines, but it's a dangerous game to start including a lot of js in html files.

ErikSearle commented on app/views.py:
```diff
@@ -5,3 +5,10 @@
+@views.route('/info', methods=['GET'])
+def info():
+    return render_template('info.html')
+'''
```
Why is this commented out?

ErikSearle commented on .gitignore:
```diff
@@ -6,3 +6,7 @@ skema_venv/*
+
+
+#Exclude venv folder
+venv/*
```
It looks like you named you virtualenv "env" not "venv". Can you fix this file, and then delete all of the env/* files from this PR please?

ErikSearle reviewed (changes requested):
Looks like a very good start to the UI design. If you haven't already, you should start taking a look at jinja and use it to make your life easier when building HTML pages.

GraemeRMorgan committed:
Local Dependencies and UI Frame

The primary UI has been integrated. I have a dummy table in place until an implementation is integrated for parsing a .csv into a table. The UI stuff can be touched up after that has been sorted.

ErikSearle reviewed (approved):
Looks good to me. Thanks for fixing the venv stuff and downloading all the files.

ErikSearle commented:
Before you merge, could you also get rid of the ds store files that mac creates?

GraemeRMorgan committed:
Added .DS_store to .gitignore

GraemeRMorgan committed:
Deleted .DS_store

All .ds_store files removed
